apm-lambda-extension/app/run.go

```diff
@@ -201,9 +207,12 @@ func (app *App) processEvent(
 	// 2) [Backup 1] RuntimeDone is triggered upon reception of a Lambda log entry certifying the end of the execution of the current function
 	// 3) [Backup 2] If all else fails, the extension relies of the timeout of the Lambda function to interrupt itself 100 ms before the specified deadline.
 	// This time interval is large enough to attempt a last flush attempt (if SendStrategy == syncFlush) before the environment gets shut down.
+
+	if app.apmClient.HasPendingFlush() {
```

Comment:
Doesn't this mean that if the the agent sends a `...?flushed=true` request to the extension a bit too late (i.e. after this call to HasPendingFlush), then the extension will wait until the runtimeDone or timeout events have occurred? I don't think that's desirable - the extension is meant to continue as soon as it can, to minimise latency overhead.

Reply:
Good catch! I've tried a different solution, let me know what you think!